### ISSUE TITLE
Boost program options broken on dev

### ISSUE THREAD

**Issue by briadam** [libqueso/queso#655]
The dev branch at 902e670 doesn't currently build with `--enable-boost-program-options`.

It's not as simple as adding the explicit instantiations.  The one for getOption is easy enough, but the ones for registerOption requires being able to insert a std::vector into an ostream for lexical cast.  I'm not sure how or even if you want to fix this...

```
./.libs/libqueso.so: undefined reference to `void QUESO::BoostInputOptionsParser::registerOption<std::vector<double, std::allocator<double> > >(std::string const&, std::vector<double, std::allocator<double> > const&, std::string const&)'
./.libs/libqueso.so: undefined reference to `void QUESO::BoostInputOptionsParser::getOption<std::vector<bool, std::allocator<bool> > >(std::string const&, std::vector<bool, std::allocator<bool> >&) const'
./.libs/libqueso.so: undefined reference to `void QUESO::BoostInputOptionsParser::registerOption<std::vector<bool, std::allocator<bool> > >(std::string const&, std::vector<bool, std::allocator<bool> > const&, std::string const&)'
```

**Comment by briadam** [comment]
I was able to track it down to around 76b72194, where the problem is with calling `registerOption` and `getOption` on `std::vector<double>` and `std::vector<bool>`.  `registerOption` fails first due to missing explicit instantiation, but then further due to inability to lexical_cast: https://github.com/libqueso/queso/blob/902e6705123d990df12dbc64c2bd2369f7a13a9e/src/gp/src/GPMSAOptions.C#L332-L370

`getOption` I think will work for both once an explicit instantiation is added for `vector<bool>`, though I'm not positive of the runtime behavior.

Since this is a new feature, and we don't intend to support BPO after say next week, I'd propose just not exposing these options in the Boost parser (commenting out even) and then removing all that code immediately post v0.58.0.


For posterity, bisect thought the offending commit was fd724a2baa76b423776b748f5cd07126c0caa720, but my test script was probably too fragile, as that wasn't quite right.  Here's the full dump:

git bisect start
git bisect bad 902e6705123d990df12dbc64c2bd2369f7a13a9e
git bisect good dd4ba59fb12cd723846dc6ee70e856894fb2db4d
git bisect good b48f4cef8a91f01ba537118d149e5720947b0f61
git bisect good d267a62ec80a70dfdd366109ff542b1fc19ea529
git bisect good 93b2283f829fea37ada41fa35358088b657c105a
git bisect bad 1e10e57e44e886e5077a8e3fecb7593d9703c653
git bisect bad 7c92e18063ca8da0d35bda8281216da0145b28c0
git bisect good 58960644288b0c84a93761f6111e0ed9fdbbc90a
git bisect good 51eeaea14775530b6da59c3055cb3d81f3329b56
git bisect bad fd724a2baa76b423776b748f5cd07126c0caa720
git bisect good a06a459be9b529199cb69226474ccc2793d53c2a

first bad commit: [fd724a2baa76b423776b748f5cd07126c0caa720] Build new GPMSA mesh code

**Comment by roystgnr** [comment]
I *thought* I'd be able to fix this quickly, and the vector<double> case is easy enough with the string conversion workaround we have elsewhere in QUESO, but I can't get vector<bool> to work at all.

I'll disable it, I think, since we're not yet supporting the periodic discrepancy basis option in any case, then by the time we add support we'll have gotten rid of program_options already.

**Comment by briadam** [comment]
I shared your optimism, but was similarly beaten down... Sounds good.

**Comment by roystgnr** [comment]
Fixed by #656
